File: xamoom-pingeborg-ios/xamoom-pingeborg-ios/PingebAnnotation.m
//
//  PingebAnnotation.m
//  xamoom-pingeborg-ios
//
//  Created by Raphael Seher on 19.03.15.
//  Copyright (c) 2015 xamoom GmbH. All rights reserved.
//

#import "PingebAnnotation.h"
#import <MapKit/MapKit.h>

@implementation PingebAnnotation

@synthesize coordinate;
@synthesize title;

- (id)initWithLocation:(CLLocationCoordinate2D)coord {
    self = [super init];
    if (self) {
        coordinate = coord;
    }
    
    return self;
}

- (id)initWithName:(NSString*)name location:(CLLocationCoordinate2D)coord {
    self = [super init];
    if (self) {
        coordinate = coord;
        title = name;
    }
    
    return self;
}

- (MKMapItem*)mapItem {
    MKPlacemark *placemark = [[MKPlacemark alloc] initWithCoordinate:coordinate addressDictionary:nil];

    MKMapItem *mapItem = [[MKMapItem alloc] initWithPlacemark:placemark];
    mapItem.name = self.title;
    
    return mapItem;
}

@end

File: xamoom-pingeborg-ios/xamoom-pingeborg-ios/views/PingebAnnotation.swift
//
//  PingebAnnotation.swift
//  pingeb.org
//
//  Created by Raphael Seher on 20/06/16.
//  Copyright © 2016 xamoom GmbH. All rights reserved.
//

import UIKit
import CoreLocation

class PingebAnnotation: NSObject, MKAnnotation {

  let title: String?
  let subtitle: String?
  let spot : XMMSpot
  let coordinate : CLLocationCoordinate2D
  var distance : Double = 0.0

  init(spot : XMMSpot, userLocation : CLLocation) {
    self.spot = spot
    self.title = spot.name
    self.coordinate = CLLocationCoordinate2D.init(latitude: spot.latitude,
                                                  longitude: spot.longitude)
    self.distance = userLocation.distance(from: CLLocation.init(latitude: spot.latitude,longitude: spot.longitude))
    
    if (distance < 1000.0) {
      self.subtitle = String.localizedStringWithFormat(NSLocalizedString("map.annotation.distance", comment: ""), distance, NSLocalizedString("map.annotation.meter", comment: ""))
    } else {
      self.subtitle = String.localizedStringWithFormat(NSLocalizedString("map.annotation.distance", comment: ""), distance/1000, NSLocalizedString("map.annotation.kilometer", comment: ""))
    }
    
    super.init()
  }
  
  func calcuateDistance(_ pointLocation: CLLocation, userLocation: CLLocation) -> Double {
    return userLocation.distance(from: pointLocation)
  }
  
}
